### ISSUE TITLE
include_background=False, softmax=True doesn't work properly in DiceFocalLoss for binary segmentation

### ISSUE THREAD

**Issue by jwc-rad** [Project-MONAI/MONAI#5413]
**Environment**
monai 1.0.1

**Describe the bug**
When using `include_background=False, softmax=True` in DiceFocalLoss for binary segmentation, the Dice part produces this warning and does not apply softmax when calculating the Dice loss:
```
UserWarning: single channel prediction, `softmax=True` ignored.
  warnings.warn("single channel prediction, `softmax=True` ignored.")
```

**To Reproduce**
```
import torch
from monai.losses import DiceFocalLoss
from monai.networks import one_hot

tx = torch.rand(1, 2, 256, 256) # logits
ty = one_hot((torch.rand(1, 1, 256, 256) > 0.5).long(), 2) # one-hot label
loss = DiceFocalLoss(include_background=False, softmax=True)
loss(tx, ty)
```

**Expected behavior & Additional context**
By `include_background=False`, since the first channel is removed before passing the input and target to Dice and Focal losses, the Dice loss cannot apply softmax to the input which now has only one channel.
https://github.com/Project-MONAI/MONAI/blob/f407fcce1c32c050f327b26a16b7f7bc8a01f593/monai/losses/dice.py#L849-L860

**Comment by wyli** [comment]
that's a good point, I submit a PR to rectify it https://github.com/Project-MONAI/MONAI/pull/5416
